### PR TITLE
Fix random resource selection.

### PR DESCRIPTION
**Pull request by f-laurens**
Disambiguate specification in `PROTOCOL.md` and fix implementation in `nabd/resources.py`:
random selection will be done **inclusively** across locale specific (if it exists) and non localized resource directories.

### CODE REVIEW

File: nabd/resources.py [hunk 1]
```diff
@@ -74,7 +76,11 @@ async def _find_random(type, parent, pattern):
...
             ]:
                 if path.is_dir():
-                    list = path.glob(pattern)
-                    if list != []:
-                        return random.choice(sorted(list))
+                    filelist = filelist + list(path.glob(pattern))
+        # logging.debug(
+        #     f"Possible selections for {str(type)} resource {str(pattern)}"
+        #     f" in {str(parent)}: {str(filelist)}"
+        # )
```
Comment (pguyot) [inline nabd/resources.py:83]: Could we just not check in commented code?
Also a test would be welcome here as everything in resources.py seems tested, but there is no use case except with #125

Comment (f-laurens) [reply nabd/resources.py:83]: > Could we just not check in commented code?

Will remove it. It was used to do some manual testing with my [nabaztag](https://github.com/f-laurens/pynab_cli/tree/release#nabaztag) script.

> Also a test would be welcome here as everything in resources.py seems tested, but there is no use case except with #125

Any quick insights on the framework/method for writing Pynab tests?

Comment (pguyot) [reply nabd/resources.py:83]: Tests in pynab are based on pytest.
Tests for `resources` module can be found here:
https://github.com/nabaztag2018/pynab/blob/master/nabd/tests/resources_test.py

Comment (f-laurens) [reply nabd/resources.py]: Hmmm.... Not easy to craft a deterministic test for a use case involving a resource existing **both** in a localized and a non localized directory...
But at least the fact that the existing `test_find_random_localized` and `test_find_random_not_localized` tests still pass shows that the fix does not introduce a regression.

Comment (pguyot) [reply nabd/resources.py]: There is no useful pattern yet with both localized and non-localized resources, there will be with PR #125 and we can draft the test then. That's why I was happy to merge your PR as is!
Testing functions with random is doable with several techniques:
- using [`random.seed`](https://docs.python.org/3/library/random.html#random.seed) to get a reproduceable sequence of numbers (probably the easiest)
- patching random using [`unittest.mock.patch`](https://docs.python.org/dev/library/unittest.mock.html#unittest.mock.patch) decorator, in our case we would assert random.choice is called with both localized and non-localized resources
- repeated calls

**Pull request merged**